### DISCUSSION TITLE
How to define a cyclic loading in BC block

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#22302] [Q&A Modules: Solid mechanics] [answered]
Dear Moose experts or users,

How to define a cyclic loading in BC block? Any developed examples? Thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello 

You can use a function to specify a known time-varying displacement. 
https://mooseframework.inl.gov/source/bcs/FunctionDirichletBC.html
Not sure about examples, you should search the repository for tensor mechanics inputs doing this.  

Guillaume
